mobile/lib/services/auth.service.dart

```diff
@@ -95,4 +129,59 @@ class AuthService {
       rethrow;
     }
   }
+
+  Future<String?> setOpenApiServiceEndpoint() async {
+    final enable = _authRepository.getEndpointSwitchingFeature();
+    if (!enable) {
+      return null;
+    }
+
+    final wifiName = await _networkService.getWifiName();
+    final savedWifiName = _authRepository.getPreferredWifiName();
+    String? endpoint;
+
+    if (wifiName == savedWifiName) {
+      endpoint = await _setLocalConnection();
+    }
+
+    endpoint ??= await _setRemoteConnection();
+
+    return endpoint;
+  }
+
+  Future<String?> _setLocalConnection() async {
+    try {
+      final localEndpoint = _authRepository.getLocalEndpoint();
+      if (localEndpoint != null) {
+        await _apiService.resolveAndSetEndpoint(localEndpoint);
+        return localEndpoint;
+      }
+    } catch (error, stackTrace) {
+      _log.severe("Cannot set local endpoint", error, stackTrace);
+    }
+
+    return null;
+  }
+
+  Future<String?> _setRemoteConnection() async {
+    try {
```

Comment:
This outer try/catch will only catch errors from `getExternalEndpointList`, so change it to only cover that.